nixos/modules/services/security/hockeypuck.nix

```diff
@@ -0,0 +1,73 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.services.hockeypuck;
+
+  configFile = pkgs.writeText "hockeypuck.config" ''
+    [hockeypuck]
+    loglevel="INFO"
+    logfile="/var/log/hockeypuck/hockeypuck.log"
+    indexTemplate="${pkgs.hockeypuck-web}/share/templates/index.html.tmpl"
+    vindexTemplate="${pkgs.hockeypuck-web}/share/templates/index.html.tmpl"
+    statsTemplate="${pkgs.hockeypuck-web}/share/templates/stats.html.tmpl"
+    webroot="${pkgs.hockeypuck-web}/share/webroot"
+
+    [hockeypuck.hkp]
+    bind=":${toString cfg.port}"
+
+    [hockeypuck.openpgp.db]
+    driver="postgres-jsonb"
+    dsn="${cfg.postgresDsn}"
+  '';
+
+in {
+  meta.maintainers = with lib.maintainers; [ etu ];
+
+  options.services.hockeypuck = {
+    enable = lib.mkEnableOption "Hockeypuck OpenPGP Key Server";
+
+    port = lib.mkOption {
+      default = 11371;
+      type = lib.types.ints.u16;
+      description = "HKP port to listen on.";
+    };
+
+    postgresDsn = lib.mkOption {
+      default = "database=hkp host=/var/run/postgresql sslmode=disable";
```

Comment:
Since you set a default I think it makes sense to add some text to the `description` here that the user is responsible for provisioning a database.